tensorflow/compiler/xla/service/gpu/ir_emitter_unnested.cc

```diff
@@ -1987,21 +2033,28 @@ void IrEmitterUnnested::EmitTile(
              llvm::Value* y_loc =
                  b_.CreateAdd(thread_id_info.thread_id_y,
                               b_.CreateMul(y_indvar, num_threads_y));
-             for (int64 j = 0; j < x_num_steps; j++) {
-               llvm::Value* x_loc =
-                   b_.CreateAdd(constant(j * step_x), start_offset_x, "x_loc");
-               IrArray::Index source_idx_x =
-                   source_idx.AddOffsetToDim(y_loc, kDimY, &b_)
-                       .AddOffsetToDim(constant(j * step_x), kDimX, &b_);
-               auto emit_element = [&] {
-                 return emit_elem_function(source_idx_x, y_loc, x_loc, j);
-               };
-               if (!x_tile_fits) {
-                 ksl->If(loop_name + "_x_in_tile",
-                         b_.CreateICmpULT(x_loc, tile_width), emit_element);
-               } else {
-                 emit_element();
-               }
+             auto unroll = [&](bool add_index_boundary_condition) {
+               return Unroll(add_index_boundary_condition, x_num_steps, step_x,
+                             vector_size, loop_name, ksl, start_offset_x, y_loc,
+                             tile_width, source_idx, b_, &emit_elem_function);
+             };
+
+             // Only try this path when we try to vectorize the loads.
+             // Special case when the tile doesn't fit completly for even row
+             // size. For odd row size every other row isn't aligned, so can't
+             // be vectorized this way by LLVM.
+             if (!x_tile_fits &&
+                 mapping_scheme.GetIndexingOrder() == kLinearStridedIndexingX) {
+               ksl->If(loop_name + "_is_full_tile",
+                       // if (block fully fit) {fast path} else {slow path}
```

Comment:
I would remove this line entirely, I think "fast path"/"slow path" is somewhat misleading, as we don't get vectorization at all on the slow path.